[Tests] Improve typing in Python tests and fix test suite dry-run execution

MarekPikula opened the pull request:
#### Summary

This pull request primarily adds and fixes type annotations and improves tying in the `scripts/tests/chiptest` and run_test_suite.py. It also expands mypy type checking coverage in the CI workflow to the affected files.

In addition, the PR fixes a bug when running the testu suite in dry-run would result in error, uncovered because of static type checks.

#### Related issues

Continuation of work on concurrent tests (#42004) – related to:
- #42261
- #42264
- #42378
- #42379
- #42380
- #42392 
- #42393

#### Testing

Tested in CI (https://github.com/MarekPikula/connectedhomeip/pull/1) and locally on Linux with run_test_suite.py.

#### Readability checklist

The checklist below will help the reviewer finish PR review in time and keep the
code readable:

-   [x] PR title is
        [descriptive](https://project-chip.github.io/connectedhomeip-doc/contributing/pull_request_guidelines.html#title-formatting)
-   [x] Apply the
        [_“When in Rome…”_](https://project-chip.github.io/connectedhomeip-doc/style/CODING_STYLE_GUIDE.html)
        rule (coding style)
-   [x] PR size is short
-   [x] Try to avoid "squashing" and "force-update" in commit history
-   [x] CI time didn't increase (typing improvements don't change runtime)

See: [Pull Request Guidelines](https://project-chip.github.io/connectedhomeip-doc/contributing/pull_request_guidelines.html)

MarekPikula committed:
Fix function visibility in IsolatedNetworkNamespace for tests

Before it was improperly marked as a private function even though it was
called from a public context.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Reduce redundant code in namespace command runner

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Fix running tests with dry run flag

After 2edb197 dry run would fail, as it is no longer a list.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Improve typing in chiptest module

Code passes mypy check and mostly Pylance strict mode.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Add ruff and mypy to suggested VS Code extensions

Also, add mypy type checker configuraton. The same which is used in CI.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Ignore missing type stubs for Pyright/Pylance

It follows the rule already applied in mypy. Maked working with strict
type checking in VS Code more pleasant.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=42204) <br/>All committers have signed the CLA.

MarekPikula commented on scripts/tests/run_test_suite.py:
```diff
@@ -60,22 +62,26 @@ class RunContext:
 
     # If not empty, exclude tests tagged with these tags
-    exclude_tags: set(TestTag) = field(default_factory={})
+    exclude_tags: set[TestTag] = field(default_factory=set[TestTag])
```
Same as above, true in Python <3.9. We use >=3.11 so generic type aliases resolve to GenericAlias.

MarekPikula commented on scripts/tests/run_test_suite.py:
```diff
@@ -60,22 +62,26 @@ class RunContext:
 
     # If not empty, include only the specified test tags
-    include_tags: set(TestTag) = field(default_factory={})
+    include_tags: set[TestTag] = field(default_factory=set[TestTag])
```
True in Python <3.9. We use >=3.11 so generic type aliases resolve to GenericAlias.

MarekPikula commented on scripts/tests/run_test_suite.py:
```diff
@@ -60,22 +62,26 @@ class RunContext:
 
     # If not empty, exclude tests tagged with these tags
-    exclude_tags: set(TestTag) = field(default_factory={})
+    exclude_tags: set[TestTag] = field(default_factory=set[TestTag])
```
True in Python <3.9. We use >=3.11 so generic type aliases resolve to GenericAlias.

MarekPikula commented on scripts/tests/chiptest/__init__.py:
```diff
@@ -318,19 +318,19 @@ def tests_with_command(chip_tool: str, is_manual: bool):
         result.check_returncode()
 
-    test_tags = set()
+    test_tags: set[TestTag] = set()
```
Not redundant for Pylance.

MarekPikula commented on scripts/tests/chiptest/__init__.py:
```diff
@@ -368,7 +368,7 @@ def _AllFoundYamlTests(treat_repl_unsupported_as_in_development: bool, treat_dft
             continue
 
-        tags = set()
+        tags: set[TestTag] = set()
```
Not redundant for Pylance.

MarekPikula committed:
Set VS Code Python editor ruler to proper line length

The same as for linters.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Fix RunSubprocess not logging full command

After introduction of SubprocessInfo, the logger started to print the
entire object repr instead of the path itself.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Add TOML extension to VS Code recommendations

TOML format is used for pyproject definition.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Move mypy configuration to pyproject

As mypy will be used wider in the project it makes sense to move the
configuration to top level.

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Fix spelling in run_test_suite help message

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

arkq commented on .github/workflows/mypy-validation.yml:
```diff
@@ -20,10 +20,15 @@ on:
       - 'src/controller/python/matter/ChipDeviceCtrl.py'
       - 'src/python_testing/matter_testing_infrastructure/**/*.py'
+      - 'scripts/tests/chipyaml/paths_finder.py'
+      - 'scripts/tests/chiptest/**/*.py'
```
Why 'scripts/tests/chiptest/**/*.py' is not in the pull_request>paths ?

MarekPikula replied:
Good catch

arkq commented on scripts/tests/chiptest/test_definition.py:
```diff
@@ -508,8 +522,8 @@ def Run(self, runner, apps_register, apps: ApplicationPaths, pics_file: str,
                 if dry_run:
-                    log.info(shlex.join(pairing_cmd))
-                    log.info(shlex.join(test_cmd))
+                    log.info(shlex.join(pairing_cmd.to_cmd()))
```
I would go for a bigger change and rename the `pairing_cmd` because `pairing_cmd.to_cmd()` looks buggy (why converting to command if it is already a command?).

MarekPikula replied:
I would say that this is not in scope of this PR, and should be tackled separately, especially with https://github.com/project-chip/connectedhomeip/pull/42129 in the pipeline.

MarekPikula commented on scripts/tests/chiptest/linux.py:
```diff
@@ -294,36 +296,36 @@ class WpaInterface(sdbus.DbusInterfaceCommonAsync,
 
         @sdbus.dbus_method_async("o")
-        async def RemoveNetwork(self, path):
+        async def RemoveNetwork(self, path: Any):
```
Is `dict[str, tuple[str, Any]]` good enough for `s{sv}`, or you want me to be more precise?

MarekPikula committed:
Add proper DBus type hints for WpaSupplicantMock

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Remove dead assignment from AppsRegister

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Fix DBus type hint

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Merge remote-tracking branch 'upstream/master' into test-typing

MarekPikula committed:
Merge remote-tracking branch 'upstream/master' into test-typing

enkiusz commented on .vscode/settings.json:
```diff
@@ -197,7 +197,8 @@
     "[python]": {
-        "editor.defaultFormatter": "ms-python.autopep8"
+        "editor.defaultFormatter": "ms-python.autopep8",
+        "editor.rulers": [132]
```
We can move this to the global section to apply to all file types.

MarekPikula committed:
Use the 132 editor ruler as the default

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Merge remote-tracking branch 'upstream/master' into test-typing

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Merge remote-tracking branch 'upstream/master' into test-typing

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Merge remote-tracking branch 'upstream/master' into test-typing

MarekPikula committed:
Add missing return annotations

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Don't use type alias default_factory

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Merge remote-tracking branch 'upstream/master' into test-typing

MarekPikula committed:
Merge remote-tracking branch 'upstream/master' into test-typing

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>

MarekPikula committed:
Merge remote-tracking branch 'upstream/master' into test-typing

Signed-off-by: Marek Pikuła <m.pikula@partner.samsung.com>
